crates/worker/src/partition/mod.rs

```diff
@@ -342,18 +359,24 @@ async fn is_outdated_or_duplicate(
 
 struct LogReader {
     log_reader: LogReadStream,
+    log_id: LogId,
 }
 
 impl LogReader {
     fn new(bifrost: &Bifrost, log_id: LogId, lsn: Lsn) -> Self {
         Self {
             log_reader: bifrost.create_reader(log_id, lsn),
+            log_id,
         }
     }
 
     async fn read_next(&mut self) -> anyhow::Result<(Lsn, Envelope)> {
+        let start = Instant::now();
         let LogRecord { record, offset } = self.log_reader.read_next().await?;
-        Self::deserialize_record(record).map(|envelope| (offset, envelope))
+        let res = Self::deserialize_record(record).map(|envelope| (offset, envelope));
+        histogram!(PP_LOG_READ_NEXT_DURATION, "log_id" => self.log_id.to_string())
+            .record(start.elapsed());
```

Comment:
I guess this metric is a lower bound for reading the next record duration since we might have already spent a bit of time reading and then another arm of the outer select loop becomes ready. In this case, we will forget about the previous wait time. Maybe we pass in the `iteration_start` instant to not forget about previous attempts?

Reply:
Same as above.